platform-emf/src/main/java/com/softicar/platform/emf/page/EmfPageBadge.java

```diff
@@ -0,0 +1,141 @@
+package com.softicar.platform.emf.page;
+
+import com.softicar.platform.dom.refresh.bus.IDomRefreshBusEvent;
+import com.softicar.platform.dom.refresh.bus.IDomRefreshBusListener;
+import com.softicar.platform.emf.validation.result.EmfDiagnosticLevel;
+import java.util.Objects;
+import java.util.function.Predicate;
+import java.util.function.Supplier;
+
+/**
+ * Defines a badge on a {@link IEmfPage} navigation link.
+ *
+ * @author Oliver Richers
+ */
+public class EmfPageBadge {
+
+	private final EmfDiagnosticLevel level;
+	private final Supplier<Integer> countSupplier;
+	private Predicate<IDomRefreshBusEvent> refreshPredicate;
+	private boolean hideZero;
+
+	/**
+	 * Constructs this badge with the given {@link EmfDiagnosticLevel} and count
+	 * supplier.
+	 *
+	 * @param level
+	 *            the {@link EmfDiagnosticLevel} (never <i>null</i>)
+	 * @param countSupplier
+	 *            the {@link Supplier} for the badge counter (never <i>null</i>)
+	 */
+	public EmfPageBadge(EmfDiagnosticLevel level, Supplier<Integer> countSupplier) {
```

Comment:
We currently do not tell the user what the displayed number refers to, in any way. When the page is opened, this might not be obvious.
Instead of letting users guess, I think we should require an `IDisplayString`-based explanatory text as an additional parameter.